codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java

```diff
@@ -587,7 +591,7 @@ private State readHeaders(ByteBuf buffer) {
 
         State nextState;
 
-        if (isContentAlwaysEmpty(message)) {
+        if (isContentAlwaysEmpty(message) || HttpUtil.hasEmptyContent(message)) {
```

Comment:
I think it is correct to assume that if this field is not present, there's no content unless we're dealing with web sockets. If content-length is set to 0 we expect only double EOLs at the end of a message which is handled nicely by this part of code.